### PR TITLE
Add Goreleaser support for releases and homebrew

### LINKED ISSUE #27: Installation without depending on go get
Currently kubecolor can be installed via `go get`, but ideally it should not be depending on go command. We should:
* enable kubecolor to be installed via package manager (homebrew etc)
* distribute pre-build binary via GitHub release

### PR DESCRIPTION
**Pull request by rothgar**
## WHAT
This adds a config for goreleaser to manually create releases via `goreleaser release` and also automatically via github actions when a new tag is pushed. I have an example in my fork and tap repo. `brew install rothgar/tap/kubecolor` works.

You'll need to create a github repo called `dty1er/homebrew-tap/` and you'll need to create a GITHUB_TOKEN with access to push into that repo. You can read more about it here https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow#using-the-github_token-in-a-workflow

You can create the token at https://github.com/settings/tokens It should only need repo access. If you want to use goreleaser to publish .rpm and .deb packages or docker containers you can also add package access.

You should make sure the token is set up in the repo settings as GITHUB_TOKEN so goreleaser can use it automatically.

Let me know if you run into problems or have questions. :+1: 

## WHY
So releases are automatic and `brew install` works

## Related issue
Fixes #27

### CODE REVIEW

**Comment by dty1er** [comment]
@rothgar 
Hi, thanks a lot for sending this PR!
I'll try it out and make sure if it works as we expected.

File: .goreleaser.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  format: tar.gz
+  format_overrides:
+    - goos: windows
+      format: zip
+
+brews:
+- name: kubecolor
+  tap:
+    owner: dty1er
+    name: homebrew-tap
```
Comment (dty1er) [inline .goreleaser.yml:36]: 📝 https://github.com/dty1er/homebrew-tap

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          login-server: docker.pkg.github.com
+          username: dty1er
+          password: ${{ secrets.GO_RELEASER }}
+      - name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v1
+        with:
+          version: latest
+          args: release --rm-dist
+        env:
+          GITHUB_TOKEN: ${{ secrets.GO_RELEASER }}
```
Comment (dty1er) [inline .github/workflows/release.yml:30]: Added a secret `GO_RELEASER`

image

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  goreleaser:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+      - name: Set up Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.13.x
+      - name: Docker login
```
Comment (dty1er) [inline .github/workflows/release.yml:18]: @rogh

Why do we need to do Docker login?

Comment (rothgar) [reply .github/workflows/release.yml:18]: oh woops, that was from another project. You don't. It's only if you want to create container images and push them to a registry. I'll remove.

Comment (dty1er) [reply .github/workflows/release.yml]: Thank you!

**Comment by dty1er** [comment]
@rothgar 
Thank you for the PR!
I will merge this and make sure this works.

**Review by dty1er — APPROVED** [review]

**Comment by dty1er** [comment]
After running below, the GH action started: https://github.com/dty1er/kubecolor/runs/1410458503

```
git tag v0.0.7
git push origin v0.0.7
```

**Comment by dty1er** [comment]
It successfully finished: https://github.com/dty1er/kubecolor/runs/1410458503

**Comment by dty1er** [comment]
v0.0.7 release is made!
https://github.com/dty1er/kubecolor/releases/tag/v0.0.7

- [x] Changelog
- [x] Binary for each environment

Also, I downloaded the build binary for Darwin and made sure it works.

**Comment by dty1er** [comment]
And homebrew-tap: https://github.com/dty1er/homebrew-tap/blob/master/Formula/kubecolor.rb 

The formula is made.

**Comment by dty1er** [comment]
I tried `brew install dty1er/tap/kubecolor` after removed current `kubecolor` under $GOPATH/bin.

```
~  $ type kubecolor
kubecolor is hashed (/usr/local/bin/kubecolor)

~  $ brew list kubecolor
/usr/local/Cellar/kubecolor/0.0.7/bin/kubecolor

~  $ brew info kubecolor
dty1er/tap/kubecolor: stable 0.0.7
Colorize your kubectl output
https://github.com/dty1er/kubecolor
/usr/local/Cellar/kubecolor/0.0.7 (5 files, 2.1MB) *
  Built from source on 2020-11-17 at 14:43:11
From: https://github.com/dty1er/homebrew-tap/blob/HEAD/Formula/kubecolor.rb
~  $ 
```

Made sure `kubecolor` installed via brew also works!

**Comment by dty1er** [comment]
@rothgar 

Thank you for your effort!
Now kubecolor gets much easier to install!

**Pull request merged**